pkg/controller/usersignup/usersignup_controller_test.go

```diff
@@ -1384,6 +1704,39 @@ func TestUserSignupDeactivatingWhenMURExists(t *testing.T) {
 		require.NoError(t, err)
 		require.Len(t, murs.Items, 0)
 		AssertThatCounterHas(t, 1)
+
+		// There should not be a notification created yet, only the next reconcile (with deleted mur) would create the notification
+		notifications := &v1alpha1.NotificationList{}
+		err = r.client.List(context.TODO(), notifications)
+		require.NoError(t, err)
+		require.Len(t, notifications.Items, 0)
+
+		// 2nd reconcile should handle creating the deactivation notification
+		res, err := r.Reconcile(req)
+		require.NoError(t, err)
+		require.Equal(t, reconcile.Result{}, res)
+
+		// lookup the userSignup and check the conditions
+		err = r.client.Get(context.TODO(), types.NamespacedName{Name: userSignup.Name, Namespace: req.Namespace}, userSignup)
+		require.NoError(t, err)
+
+		// Confirm the status has been set to Deactivated and the deactivation notification is created
+		test.AssertConditionsMatch(t, userSignup.Status.Conditions,
+			v1alpha1.Condition{
+				Type:   v1alpha1.UserSignupApproved,
+				Status: v1.ConditionTrue,
+				Reason: "ApprovedAutomatically",
+			},
+			v1alpha1.Condition{
+				Type:   v1alpha1.UserSignupComplete,
+				Status: v1.ConditionTrue,
+				Reason: "Deactivated",
+			},
+			v1alpha1.Condition{
+				Type:   v1alpha1.UserSignupUserDeactivatedNotificationCreated,
+				Status: v1.ConditionTrue,
+				Reason: "NotificationCRCreated",
+			})
```

Comment:
Could you please move this into a subtest? This verifies another functionality based on the previous test - it's much easier to read when it's used as a subtest.

Reply:
Sure, makes sense!

Reply:
Done in https://github.com/codeready-toolchain/host-operator/pull/289/commits/a546d2527c71eb6962eb1428bc6ec22e8f6ceca9